### ISSUE TITLE
Implement .detach() on Q nodes

### ISSUE THREAD

**Issue by masak** [masak/007#62]
Proposing a way to make a Qtree lose its original context.

    macro so_hygienic() {
        my x = "yay, clean!";
        return quasi { say(x) };
    }

    macro so_unhygienic() {
        my x = "yay, clean!";
        return quasi { say(x) }.detach();
    }

    my x = "gross!";
    so_hygienic();    # yay, clean!
    so_unhygienic();  # gross!

In other words, context-lessness would no longer be just the domain of synthetic Qtrees, but also Qtrees that you thus detached.

It is unclear to me how deep the detaching should reach into the Qtree. If a Qtree is mostly organic, say, but it has some synthetic bits here and there, and those synthetic bits contain organic sub-bits...

    organic
        synthetic
            organic

...would it be more intuitive for the Qtree to lose *all* its context, or just the topmost organic context?

**Comment by vendethiel** [comment]
interesting. thought: shouldn't this be an adverb?

**Comment by masak** [comment]
> interesting. thought: shouldn't this be an adverb?

On quasis that would make some sense, yes &mdash; the adverb could attach to the quasi keyword somehow, perhaps. It would make less sense on variables containing Qtrees, for example macro arguments.

Also, 007 doesn't currently have adverbs. 😛
